viewing_party/party.py

```diff
@@ -1,23 +1,231 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+
+    if title == False or title == None:
+        return None
+    if genre == False or genre == None:
+        return None
+    if rating == False or rating == None:
+        return None
+
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    if movie is not False:
+        user_data["watched"].append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    if movie is not False:
+        user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+    to_watch = len(user_data["watchlist"])
+    movie_index = 0
+    
+    while movie_index < to_watch:
+        if user_data["watchlist"][movie_index]["title"] == title:
+            del user_data["watchlist"][movie_index]
+            user_data["watched"].append(title)
+        movie_index += 1
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    starting_avg = 0.0
+        
+    ratings_list = []
+    deets = 0
+    watched_movies = len(user_data['watched'])
+
+    while deets < watched_movies:
+        new_rating = user_data['watched'][deets]['rating']
+        ratings_list.append(new_rating)
+        deets += 1
+    
+    for x in ratings_list:
+        starting_avg = float(starting_avg + x)
+
+    final_avg = None
+
+    if len(ratings_list) == 0:
+        return starting_avg
+    else:
+        compute_avg = starting_avg/len(ratings_list)
+        return compute_avg
+
+def get_most_watched_genre(user_data):
+    if user_data['watched'] == []:

```

Comment:
This is one way of checking the truthiness of the user watched list. We can also check for it another way. Something like this:
```if not user_data['watched']```